www/base/src/app/d3/d3.service.js

```diff
@@ -10,10 +10,9 @@ class D3 {
     constructor($document, $q, config, $rootScope) {
         const d = $q.defer();
 
-        // Resolve function
-        $.getScript('d3.min.js', () =>
-            $rootScope.$apply(() => d.resolve(window.d3))
-        );
+        import('d3').then(module => {
+            d.resolve(module);
```

Comment:
the goal was to put d3 as a separate webpack bundle (as it is pretty big)
did you plan to do that?

Reply:
Thids is "dynamic" import, it should import d3 from a separate bundle only when this code is run. This is why the import statement returns a promise.